### PR TITLE
Update git.md with more details about squash/amend commits

### PR DESCRIPTION
**Pull request by vorburger**
Follow-up to https://github.com/google/android-fhir/pull/2315#discussion_r1675662809 ...

### CODE REVIEW

File: docs/contrib/git.md [hunk 1]
```diff
@@ -4,6 +4,27 @@ This page documents [Git](https://git-scm.com) tips for contributors to this pro
...
+   of this initial commit should explain what this PR is all about.
+
+1. Maintainers, or other Contributors, will now review the PR. They may add comments requesting changes.
+
+1. When maintainers update PRs to make changes requested by reviewers, those should be added as
+   additional new single commits per round of review,
+   typically with a generic message such as _"Incorporated review feedback."_
+   Do NOT squash (or amend) these review updates into the original commit.
+
+1. Maintainers (AKA Committers) _squash_ all commits of PRs into a single commit when merging.
```
Comment (bashir2) [inline docs/contrib/git.md:26]: Just FYI, in the backend repos (i.e., data-pipes and gateway) we actually disable other options to make sure that all merges are done with squash. You can do that under Settings > General:
image

**Review by bashir2 — APPROVED** [review]
Thanks for clarifying the PR workflow @vorburger.

File: docs/contrib/git.md (hunk 1 above)
Comment (jingtang10) [reply docs/contrib/git.md:26]: we have the same :)

**Comment by vorburger** [comment]
@jingtang10 my understanding, from comments above, is that you are cool with this; I'm therefore (exceptionally) "self merging" this now...

File: docs/contrib/git.md [hunk 2]
```diff
@@ -4,6 +4,27 @@ This page documents [Git](https://git-scm.com) tips for contributors to this pro
 
 This project generally follows [these guidelines](https://github.com/google/fhir-data-pipes/blob/master/doc/review_process.md) from our `fhir-data-pipes` sister project.
 
+## Overview
+
+We use the following workflow:
+
+1. Contributors can open "draft" PRs for any Work In Progress (WIP) which does not require reviews yet.
```
Comment (jingtang10) [inline docs/contrib/git.md:11]:
```suggestion
1. Contributors can open "draft" PRs for any Work In Progress (WIP) which does not require review yet.
```

File: docs/contrib/git.md [hunk 3]
```diff
@@ -4,6 +4,27 @@ This page documents [Git](https://git-scm.com) tips for contributors to this pro
...
+   It's OK if these don't fully build, yet.
```
Comment (jingtang10) [inline docs/contrib/git.md:12]:
```suggestion
   It's OK if the checks do not pass yet.
```

File: docs/contrib/git.md [hunk 4]
```diff
@@ -4,6 +4,27 @@ This page documents [Git](https://git-scm.com) tips for contributors to this pro
...
+
+1. When a PR is ready for code review, Contributors un-draft it. It must fully build green.
+   At this point, PRs should only have 1 single commit; please "squash" (or use `--amend`) your "local history",
```
Comment (jingtang10) [inline docs/contrib/git.md:15]:
```suggestion
   At this point, the PR should only have 1 single commit; please "squash" (or use `--amend`) your "local history",
```

**Review by jingtang10 — APPROVED** [review]
thank you @vorburger!

**Review by joiskash — COMMENTED** [review]
Great process. Just a nit, the fhir data pipes README has hyper links to git concepts like amend and squash etc. 
Another nit, all the points start with 1. instead of 1,2,3

**Comment by vorburger** [comment]
> Great process. 

@joiskash Thank You!

> Just a nit, the fhir data pipes README has hyper links to git concepts like amend and squash etc.

If you would find this useful, would you like to raise a PR to proposing adding those? I'm happy to review it.

> Another nit, all the points start with 1. instead of 1,2,3

That's not a problem, and "normal", in Markdown syntax, as per https://spec.commonmark.org/0.31.2/#lists, quote: _"The numbers of subsequent list items are disregarded."_ (i.e. autonumbered, by the "Markdown Processor"). 

Ergo, on https://google.github.io/android-fhir/contrib/git/ where it's "rendered" (as [described here](https://google.github.io/android-fhir/contrib/docs/#implementation-details)), it's correctly numbered, of course. I hope this helps to clarify?

**Comment by joiskash** [comment]
> > Great process.
> 
> @joiskash Thank You!
> 
> > Just a nit, the fhir data pipes README has hyper links to git concepts like amend and squash etc.
> 
> If you would find this useful, would you like to raise a PR to proposing adding those? I'm happy to review it.
Sure! 
> > Another nit, all the points start with 1. instead of 1,2,3
> 
> That's not a problem, and "normal", in Markdown syntax, as per https://spec.commonmark.org/0.31.2/#lists, quote: _"The numbers of subsequent list items are disregarded."_ (i.e. autonumbered, by the "Markdown Processor").
> 
> Ergo, on https://google.github.io/android-fhir/contrib/git/ where it's "rendered" (as [described here](https://google.github.io/android-fhir/contrib/docs/#implementation-details)), it's correctly numbered, of course. I hope this helps to clarify?
Yes it does, thank you! I was looking for an example and the documentation on this. Thanks for sharing.

**Pull request merged**